### PR TITLE
Manual OSX-ARM64

### PR DESCRIPTION
**Pull request by Tobias-Fischer**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by Tobias-Fischer** [comment]
I'm not sure what's going on with the Windows builds, and the error is unrelated to this PR:
```
Traceback (most recent call last):
  File "D:\bld\pythonocc-core_1650930655183\_test_env\lib\site-packages\OCC\Core\GeomLib.py", line 20, in swig_import_helper
    return importlib.import_module(mname)
  File "D:\bld\pythonocc-core_1650930655183\_test_env\lib\importlib\__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 657, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 556, in module_from_spec
  File "<frozen importlib._bootstrap_external>", line 1166, in create_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
ImportError: DLL load failed while importing _GeomLib: The specified procedure could not be found.
```

Is it okay to skip them for now so we can get working osx-arm64 builds @conda-forge/pythonocc-core?

**Comment by Tobias-Fischer** [comment]
Actually I found that occt needs to be specified precisely at runtime for Windows, too. Trying this now.

**Comment by Tobias-Fischer** [comment]
This is finally ready for review/merge @conda-forge/pythonocc-core

File: recipe/meta.yaml [hunk 1]
```diff
@@ -9,24 +9,26 @@ source:
...
   host:
     - python
-    - occt
+    - occt {{ version }}
     - tbb-devel
   run:
     - python
-    - occt
+    - occt                  # [unix]
+    - {{ pin_compatible('occt', max_pin='x.x.x') }}  # [win]
```
Comment (looooo) [inline recipe/meta.yaml:31]: I guess we should use this for all platforms. occt must be tightly pinned. But it seems like this syntax doesn't work (looking at the latest windows builds)

Comment (Tobias-Fischer) [reply recipe/meta.yaml]: Okay I've done the run_constrained (which works) for all platforms in the latest commit. No idea why the pin_compatible doesn't work.

**Comment by Tobias-Fischer** [comment]
Okay, this is now really ready @looooo :)

**Pull request merged**